controllers/chaosimpl/jvmchaos/impl.go

```diff
@@ -161,46 +190,57 @@ func generateRuleData(spec *v1alpha1.JVMChaosSpec) error {
 		return nil
 	}
 
-	ruleParameter := &JVMRuleParameter{
-		JVMParameter: spec.JVMParameter,
+	bytemanTemplateSpec := BytemanTemplateSpec{
+		Name:   spec.Name,
+		Class:  spec.Class,
+		Method: spec.Method,
 	}
 
 	switch spec.Action {
 	case v1alpha1.JVMLatencyAction:
-		ruleParameter.Do = fmt.Sprintf("Thread.sleep(%d)", ruleParameter.LatencyDuration)
+		bytemanTemplateSpec.Do = fmt.Sprintf("Thread.sleep(%d)", spec.LatencyDuration)
 	case v1alpha1.JVMExceptionAction:
-		ruleParameter.Do = fmt.Sprintf("throw new %s", ruleParameter.ThrowException)
+		bytemanTemplateSpec.Do = fmt.Sprintf("throw new %s", spec.ThrowException)
 	case v1alpha1.JVMReturnAction:
-		ruleParameter.Do = fmt.Sprintf("return %s", ruleParameter.ReturnValue)
+		bytemanTemplateSpec.Do = fmt.Sprintf("return %s", spec.ReturnValue)
 	case v1alpha1.JVMStressAction:
-		if ruleParameter.CPUCount > 0 {
-			ruleParameter.StressType = "CPU"
-			ruleParameter.StressValueName = "CPUCOUNT"
-			ruleParameter.StressValue = fmt.Sprintf("%d", ruleParameter.CPUCount)
+		bytemanTemplateSpec.Helper = StressHelper
+		bytemanTemplateSpec.Class = TriggerClass
+		bytemanTemplateSpec.Method = TriggerMethod
+		// the bind and condition is useless, only used for fill the template
+		bytemanTemplateSpec.Bind = "flag:boolean=true"
+		bytemanTemplateSpec.Condition = "true"
+		if spec.CPUCount > 0 {
+			bytemanTemplateSpec.Do = fmt.Sprintf("injectCPUStress(\"%s\", %d)", spec.Name, spec.CPUCount)
 		} else {
-			ruleParameter.StressType = "MEMORY"
-			ruleParameter.StressValueName = "MEMORYTYPE"
-			ruleParameter.StressValue = ruleParameter.MemoryType
+			bytemanTemplateSpec.Do = fmt.Sprintf("injectMemStress(\"%s\", %s)", spec.Name, spec.MemoryType)
```

Comment:
If CPUCount>0 and MemoryType is not empty, this will only inject MemStress, is this expected?

Reply:
I took a look and the existing logic is like this. Is it necessary to support injection of CPUStress and MEMStress at the same time in the future?

Reply:
There is a validation in the webhook, can only set one of `CPUCount` and `MemoryType`. In my opinion, it's not necessary to inject both of them in one experiment.
https://github.com/chaos-mesh/chaos-mesh/blob/b4b33740bb6dbcf93892d693adc32376f9bd5a18/api/v1alpha1/jvmchaos_webhook.go#L51